File: src/main/java/io/github/takenoha/towerdefense/persistence/PreparedRollback.java
package io.github.takenoha.towerdefense.persistence;

import java.util.Objects;
import java.util.UUID;

/** A rollback decision whose physical/database completion was interrupted. */
public record PreparedRollback(UUID operationId, BlockRollbackDecision decision) {
    public PreparedRollback {
        Objects.requireNonNull(operationId, "operationId");
        Objects.requireNonNull(decision, "decision");
    }
}

File: src/main/kotlin/io/github/takenoha/towerdefense/persistence/PreparedRollback.kt
package io.github.takenoha.towerdefense.persistence

import java.util.UUID
import kotlin.jvm.JvmRecord

/** A rollback decision whose physical/database completion was interrupted. */
@JvmRecord
data class PreparedRollback(
    val operationId: UUID,
    val decision: BlockRollbackDecision,
)
